numpy/_core/src/multiarray/alloc.c

```diff
@@ -274,10 +274,8 @@ PyDataMem_RENEW(void *ptr, size_t size)
     void *result;
 
     assert(size != 0);
+    PyTraceMalloc_Untrack(NPY_TRACE_DOMAIN, (npy_uintp)ptr);
     result = realloc(ptr, size);
-    if (result != ptr) {
-        PyTraceMalloc_Untrack(NPY_TRACE_DOMAIN, (npy_uintp)ptr);
-    }
     PyTraceMalloc_Track(NPY_TRACE_DOMAIN, (npy_uintp)result, size);
```

Comment:
Note that `PyTraceMalloc_Track` can return an error.

Reply:
Oy. [It can return](https://docs.python.org/3/c-api/memory.html#c.PyTraceMalloc_Track) 0 for success, -1 for MemoryError, -2 for tracing disabled. Good catch.